marketplace/deployer_util/provision.py

```diff
@@ -272,8 +272,10 @@ def provision_service_account(schema, prop, app_name, namespace):
         'apiVersion': 'rbac.authorization.k8s.io/v1',
         'kind': 'RoleBinding',
         'metadata': {
-            'name': '{}:{}:{}-rb'.format(app_name, prop.name, role),
-            'namespace': namespace,
+            'name':
+                dns1123_name('{}:{}:{}-crb'.format(app_name, prop.name, role)),
```

Comment:
`dns1123_name` is not the right one to use for this, as this name doesn't need to conform to DNS spec. It also turns colons into dashes. I would suggest creating a new method to cap the length of the resources, and appending a hash suffix similar to `dns1123_name` method to avoid collisions.

Still think the colon convention is good. Here is an example of what `kubectl get clusterrolebindings` returns:

```
NAME                                                   AGE
gce:beta:kubelet-certificate-bootstrap                 83d
gce:beta:kubelet-certificate-rotation                  83d
system:aws-cloud-provider                              83d
system:basic-user                                      83d
system:controller:certificate-controller               83d
system:controller:ttl-controller                       83d
system:discovery                                       83d
```